elastic/kibana
# [InfraOps] Preserve time values when navigating from the waffle map to the logs and details pages

This adds `time` and `from`/`to` parameters to the metrics and logs links of node context menu in the waffle map:

![image](https://user-images.githubusercontent.com/973741/47579737-20c09000-d94d-11e8-9374-0d74017a84d3.png)
![image](https://user-images.githubusercontent.com/973741/47579755-2d44e880-d94d-11e8-9d7a-244b79c13fa4.png)

## Implementation Notes

As part of this, the three `link-to` routes each for `(host|container|pod)-logs` and `(host|container|pod)-detail` were collapsed into one route respectively, that parses the node type from the url. The corresponding `<RedirectTo*Logs>` and `<RedirectTo*Detail>` components were unified to `<RedirectToNodeLogs>` and `<RedirectToNodeDetail>` respectively. Similarly, the `get*LogsUrl()` and `get*DetailUrl()` functions were unified into `getNodeLogsUrl()` and `getNodeDetailUrl()`.

Since now the timerange is passed from the waffle map to the detail page it became apparent that the used time ranges and intervals were different. This adjusts the waffle map values to match the default detail page values of a one hour range with a one minute "last bucket" (as described in the header text of the waffle map page).

## How to test

* navigate to the waffle map
* select a different, specific point in time, for which node data are present
* observe that the "View logs" and "View metrics" links in the individual nodes' context menu contain the `time` or `from`/`to` timestamps
* navigate to the detail pages using the above links
* observe that the selected time ranges' end point match the point in time previously selected on the waffle map
* observe that the selected time ranges are one hour in size
